src/connection.cr

```diff
@@ -21,16 +21,31 @@ module Redis
     # SSL connections require specifying the `rediss://` scheme.
     # Password authentication uses the URI password.
     # DB selection uses the URI path.
+    #
+    # To use Unix sockets: `socket:/path/to/redis.sock`
     def initialize(@uri = URI.parse("redis:///"))
-      host = uri.host || "localhost"
-      port = uri.port || 6379
-      socket = TCPSocket.new(host, port)
-      socket.sync = false
-
-      # Check whether we should use SSL
-      if uri.scheme == "rediss"
-        socket = OpenSSL::SSL::Socket::Client.new(socket)
+      if uri.scheme == "socket"
```

Comment:
Thinking about this a bit more, I think the `socket` URL scheme may be a bit too generic a name — they're _all_ sockets. 😄 Do you think it might make more sense to use `unix` as the URL scheme?

Reply:
I see the point. Unfortunately, there is not a standard among the Redis clients to build on. Redis itself (e.g. in redis.conf) refers to it as "unixsocket". Maybe that is the clearest?

It is also more consistent with the other existing Redis client for Crystal: https://github.com/stefanwille/crystal-redis/blob/009e35fa40bbd518798afcc32c397402c6c6acb2/src/redis/connection.cr#L8

Reply:
Apologies for the wildly delayed reply here. I seem to have pulled this move:

![oh no](https://user-images.githubusercontent.com/108205/120064928-ce800800-c03c-11eb-8643-4cf10aaa08c8.png)

If the Redis server recommends `unixsocket`, I imagine that's what most Redis clients also use, and consistency can be pretty important. Admittedly, since I almost always use TCP sockets for IPC even when Unix sockets are available, I don't have many personal opinions on how to use Unix sockets so my only real recommendation is least-surprise. 🙂